File: LLM10-unbounded-consumption/inference-call-without-timeout-or-token-cap/bad_openai_no_max_tokens.py
"""Customer-facing chat service.

Uses the next-generation OpenAI flagship model for the public chat endpoint.
"""

from __future__ import annotations

import os

from openai import OpenAI

CHAT_MODEL = "gpt-5.4"


def _client() -> OpenAI:
    return OpenAI(api_key=os.environ["OPENAI_API_KEY"])


def reply(user_msg: str) -> str:
    """Return the assistant reply for the given user message."""
    client = _client()
    response = client.chat.completions.create(
        model="gpt-5.4",
        messages=[
            {"role": "system", "content": "You are a helpful support assistant."},
            {"role": "user", "content": user_msg},
        ],
        temperature=0.2,
    )
    return response.choices[0].message.content

File: LLM10-unbounded-consumption/inference-call-without-timeout-or-token-cap/bad_openai_no_max_tokens.ts
/**
 * Triggers inference-call-without-timeout-or-token-cap: same pattern as the
 * Python OpenAI SDK case, ported to the JS/TS ecosystem. No `max_tokens`,
 * no per-request `timeout`, and no wrapping rate-limiter/circuit-breaker
 * around the call — a long or adversarial prompt can generate an unbounded
 * response and there is nothing here to cut it off.
 */
import OpenAI from "openai";

const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

export async function reply(userMsg: string): Promise<string> {
  const response = await openai.chat.completions.create({
    model: "gpt-4o",
    messages: [
      { role: "system", content: "You are a helpful support assistant." },
      { role: "user", content: userMsg },
    ],
  });
  return response.choices[0].message.content ?? "";
}
